src/process.rs

```diff
@@ -129,10 +129,10 @@ impl PtyProcess {
     }
 
     /// Get handle to pty fork for reading/writing
-    pub fn get_file_handle(&self) -> File {
+    pub fn get_file_handle(&self) -> Result<File, Error> {
         // needed because otherwise fd is closed both by dropping process and reader/writer
-        let fd = dup(self.pty.as_raw_fd()).unwrap();
-        unsafe { File::from_raw_fd(fd) }
+        let fd = dup(self.pty.as_raw_fd()).map_err(|err| Error::Io(err.into()))?;
```

Comment:
Is it not possible to use the "?" here? Since error::Io has a #[from] macro.

Reply:
Initially I didn't notice the function was not returning a Result and I misinterpreted the error.